ocaml/typing/typecore.ml

```diff
@@ -4354,7 +4361,8 @@ and type_expect_
         exp_attributes = sexp.pexp_attributes;
         exp_env = env }
   | Pexp_fun (l, Some default, spat, sbody) ->
-      assert(is_optional_parsetree l); (* default allowed only with optional argument *)
+      let l = Typetexp.transl_label l None in
```

Comment:
The use of `None` here is cheating a bit: you're taking advantage of the fact that an optional label always ignores the type. Maybe that's better than the alternative, but it would be helpful to comment that you're not passing the type in (which you could get by looking inside the pattern).